__tests__/app.test.js

```diff
@@ -24,141 +24,192 @@ describe("/api", () => {
+    return request(app)
+      .get("/api/topix")
+      .expect(404)
+      .then((response) => {
+        expect(response.status).toBe(404);
+        expect(response.body.message).toBe("endpoint not found!");
+      });
+  });
+});
+describe("/GET api/articles", () => {
+  test(" GET /api/articles/ returns an array", () => {
+    return request(app)
+      .get("/api/articles/")
+      .expect(200)
+      .then(({ body }) => {
+        const articles = body.articles;
+        expect(Array.isArray(articles)).toBe(true);
+      });
+  });
+  test(" GET /api/articles/ returns a status 200 and all articles information except for body", () => {
+    return request(app)
+      .get("/api/articles/")
+      .expect(200)
+      .then(({ body }) => {
+        const articles = body.articles;
+        expect(articles.length).toBe(13);
+        articles.forEach((article) => {
+          expect(article.hasOwnProperty("article_id")).toBe(true);
+          expect(article.hasOwnProperty("title")).toBe(true);
+          expect(article.hasOwnProperty("topic")).toBe(true);
+          expect(article.hasOwnProperty("author")).toBe(true);
+          expect(article.hasOwnProperty("created_at")).toBe(true);
+          expect(article.hasOwnProperty("votes")).toBe(true);
+          expect(article.hasOwnProperty("article_img_url")).toBe(true);
+          expect(article.hasOwnProperty("body")).toBe(false);
         });
-    });
-    test("GET /api/articles/banana returns a 404 and 'not found' as parameter is invalid", () => {
-      return request(app)
-        .get("/api/articles/hello")
-        .expect(404)
-        .then(({ body }) => {
-          expect(body.message).toBe("not found");
+      });
+  });
+  test(" GET /api/articles/ returns a status 200 and a comment_count property on all articles", () => {
+    return request(app)
+      .get("/api/articles/")
+      .expect(200)
+      .then(({ body }) => {
+        const articles = body.articles;
+        articles.forEach((article) => {
+          expect(article.hasOwnProperty("comment_count")).toBe(true);
         });
-    });
-    test("GET /api/articles/100 returns a 400 and error message as parameter is valid but doesn't exist in database", () => {
-      return request(app)
-        .get("/api/articles/100")
-        .expect(200)
-        .then(({ body }) => {
-          expect(body).toEqual({});
+      });
+  });
+  test(" GET /api/articles/ returns correct comment count for each article", () => {
+    return request(app)
+      .get("/api/articles/")
+      .expect(200)
+      .then(({ body }) => {
+        const articles = body.articles;
+        expect(articles[0].comment_count).toBe(2);
+        expect(articles[1].comment_count).toBe(1);
+        expect(articles[2].comment_count).toBe(0);
+        expect(articles[3].comment_count).toBe(0);
+        expect(articles[4].comment_count).toBe(0);
+        expect(articles[5].comment_count).toBe(2);
```

Comment:
I dont think this is necessary  - we could use a forEach and say it was a value of number :)